sdk/client/Cargo.toml

```diff
@@ -24,8 +24,12 @@ serde_json = "1.0.64"
 
 diem-crypto = { path = "../../crypto/crypto", version = "0.0.1" }
 diem-json-rpc-types  = { path = "../../json-rpc/types", version = "0.0.1" }
+move-core-types = { path = "../../language/move-core/types", version = "0.0.1" }
 diem-types = { path = "../../types", version = "0.0.1" }
 
+# TODO: do we need to release a 0.0.1 crate for VM?
+vm = { path = "../../language/vm"}
+
```

Comment:
Yeah :/ this crate is included in the set that is published to crates.io, which means that it can't have any dependencies which aren't published on crates.io.

There's a couple options here:
1. publish the VM on crates.io (that might be high effort since i don't know what more work that would entail, I suppose @vgao1996 might know)
2. We start a `client-ext` crate (or something with a better name) and stick this logic in there, attaching these functions to the Client and Blocking Client via extension traits until the vm crates is published on crates.io (then we can merge it back).

Least these are some potential solutions, I'm open to others :)

Reply:
My two cents:
(1) should eventually be doable. However we'll have to 
- Wait until #8161 lands so that we use the proper crate name ("move-binary-format")
- Get rid of some of the Diem dependencies (or maybe get them published to crates.io?)
  - diem-crypto
  - diem-proptest-helpers
  - num-variants

I don't know how hard it would be to remove those dependencies yet -- will probably figure it out this week as I continue working on removing these dependencies. If unsure, I'd personally go with route (2) just to be safe.

Reply:
To move forward with this PR, I'm happy to get rid of the module-related API's for now + bring them back once we get the VM crate into a publishable state.

Reply:
Fixed.